[Luah Jun Yang} iP

LuahJunYang opened the pull request:
### Frenchie
>"I need more boulets." -Frenchie

Frenchie keeps track of all tasks in your life in a fuss-free CLI. It is:
- Simple to learn
- Text-commands based
- **SUPER FAST**

All you need to start using Frenchie is:

1. Download Frenchie from [here](https://github.com/LuahJunYang/ip)
2. Double click the jar file
3. Enter your tasks in the right format
4. Let it track all of your tasks for you 👍🏻 

Features:

- [x] Task Tracking
- [ ] _Reminders for Tasks with deadlines (coming soon!)_

For those of you who code, Frenchie is written in java and you can experiment with it to add your own features!

Here is the main() method of Frenchie:
```java
public static void main(String[] args) {
        new Frenchie("frenchie.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

LuahJunYang committed:
Level 0

LuahJunYang committed:
Finished Level 1, renamed bot to Frenchie

LuahJunYang committed:
Added functionality to store user input as list and print out all input text when 'list' is input

LuahJunYang committed:
Created Task class to be used for Frenchie

LuahJunYang committed:
Added addTask and listTask methods and a constructor to Frenchie.java

LuahJunYang committed:
Added mark as incomplete method to Task.java and complete and uncomplete method to Frenchie.java, level 3 completed

LuahJunYang committed:
Created new Task subclasses

LuahJunYang committed:
Added if-else statements and updated toString() methods for checking for todo, event, deadline. Level 4 Complete.

LuahJunYang committed:
Created new FrenchieException class to handle basic exceptions for now. Level 5 completed up till basic exceptions.

LuahJunYang committed:
Added delete function to frenchie, changed tasks attribute to ArrayList instead of List

sushiyade commented on src/main/java/Frenchie.java:
```diff
@@ -79,6 +81,16 @@ public static void main(String[] args) {
                                 target_task.toString() + "\n" +
                                 "____________________________________________________________");
                     }
+                } else if (input.contains("delete")){
+                    String[] parts = input.split(" ");
+                    int index = Integer.parseInt(parts[1]) - 1;
+                    Task target_task = frenchie.tasks.get(index);
+                    frenchie.deleteTask(index);
+                    System.out.println("____________________________________________________________\n" +
+                            "Noted. I've removed this task: \n" +
+                            target_task.toString()   + "\n" +
+                            "Now you have " + frenchie.getNumOfTasks() + " tasks in the list.\n" +
+                            "____________________________________________________________");
```
As the number of keywords increase, perhaps you might want to start looking at enums to have a collection of your keywords!

yytan25 replied:
Seconded. In addition, perhaps you may wish to divide all the logic in your `main` method into several other methods, as the complexity of your program increases.

yytan25 commented on src/main/java/Frenchie.java:
```diff
@@ -0,0 +1,144 @@
+        this.tasks = new ArrayList<>();
+    }
+
+    public void addTask(Task NEW_TASK) {
```
The Task parameter is a variable, not a constant. So it should be in camelCase instead, i.e. `newTask`.

yytan25 commented on src/main/java/Frenchie.java:
```diff
@@ -0,0 +1,144 @@
+    }
+
+    public void completeTask(int index) {
+        tasks.get(index).mark_as_completed();
```
Please follow the camelCase convention for methods, i.e. `markAsCompleted`.

yytan25 commented on src/main/java/Frenchie.java:
```diff
@@ -0,0 +1,144 @@
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo); */
+        Frenchie frenchie = new Frenchie();
+        String skeleton = "____________________________________________________________\n" +
```
When breaking up a statement into multiple lines, the `+` operator should be on a new line.

yytan25 commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,11 @@
+class ToDo extends Task {
```
Perhaps you should add the `public` access modifier to your class definition?

yytan25 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,16 @@
+public class Event extends Task{
```
Should leave a whitespace between your definition and the curly braces, as in
`public class Event extends Task {`

yytan25 commented on src/main/java/Frenchie.java:
```diff
@@ -0,0 +1,144 @@
+                    break;
+                } else if (input.equals("list")) { //Checking if user is looking to list all tasks
+                    frenchie.listTasks();
+                } else if (input.contains("mark")) { //Checking if user input is to mark/unmark tasks
```
This doesn't check if the input is either `mark` or `unmark`. If a user inputs `markzzzzzz`, this will still be treated as `unmark` even though it is an invalid command, because the input still contains the substring `mark`. To avoid similar problems like this, perhaps you should use Java Enums to ensure input validity.

yytan25 reviewed (commented):
Good use of OOP and inheritance for the subclasses of `Task`. You can still improve on your coding style and your use of SLAP.

LuahJunYang committed:
Added functions saveTasksToFile and readTasksFromFile

LuahJunYang committed:
Added LocalDateTime and DateTimeFormatter functionalities to constructors for Event and Deadline classes as well as save and load methods for Frenchie

LuahJunYang committed:
Implemented enum for Commands, abstracted UI, Parser, Storage, and TaskList class

LuahJunYang committed:
Bug fixing for bye command, causing an illegal state exception when the scanner is closed

LuahJunYang committed:
Packaged all classes into package.frenchie

LuahJunYang committed:
Merge branch 'add-gradle-support' for 'A-Gradle'

Kevin-Liusx commented on src/main/java/Frenchie/Storage.java:
```diff
@@ -0,0 +1,74 @@
+                    LocalDateTime storedEndTime = LocalDateTime.parse(endTime, inputFormatter);
+                    DateTimeFormatter desiredFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm");
+                    String constructorStartTime = storedStartTime.format(desiredFormatter);
+                    String constructorEndTime = storedEndTime.format(desiredFormatter);
```
I think this part of your code and be improved? Maybe try to use some other methods?😊

LuahJunYang committed:
Updated build.gradle file, wrote J-unit tests ToDoTest and TaskTest

LuahJunYang committed:
Added JavaDoc comments for Task, ToDo, Deadline, Event, Ui, Parser classes.

LuahJunYang committed:
Added findMessage() method to Ui.java, find enum to Command, and returnMatchTasks() method to TaskList.java. Level-9 Completed.

LuahJunYang committed:
Resolved merge conflict in Ui.java

LuahJunYang committed:
Edited void functions to return String for the GUI. Implemented GUI successfully.

LuahJunYang committed:
Improving on Code Quality
Removed unused methods and corrected the order of import statements to be in the correct lexographical order. Fixed the error of tasks not loading from frenchie.txt by calling storage.load() before parsing the input String.

LuahJunYang committed:
Merge pull request #3 from LuahJunYang/branch-A-CodeQuality

Improving on Code Quality

LuahJunYang committed:
Add help function

Added help as an enum in Command and included 'help' as a case for the Parser class. Added a quick start guide that will be presented to the user when user inputs 'help' as the command.

LuahJunYang committed:
Merge pull request #4 from LuahJunYang/branch-C-Help

Add help function

LuahJunYang committed:
Added User Guide in README.md

LuahJunYang committed:
Updating README for github page

LuahJunYang committed:
Added Screenshot of UI

LuahJunYang committed:
Updated mainClass in build.gradle to fix the jar file not running
